File: data/code_pairs/fortran/1822.f90
module adios_helpers_mod
  implicit none
contains
  subroutine define_adios_scalar(scalar)
    real, intent(in) :: scalar
    print *, "Defining ADIOS scalar with value: ", scalar
  end subroutine define_adios_scalar

  subroutine define_adios_global_real_1d_array(array, size)
    real, dimension(:), intent(in) :: array
    integer, intent(in) :: size
    print *, "Defining ADIOS global real 1D array of size ", size
    print *, array
  end subroutine define_adios_global_real_1d_array

  subroutine check_adios_err()
    print *, "Checking for ADIOS errors..."
  end subroutine check_adios_err
end module adios_helpers_mod

File: data/code_pairs/cpp/1822.cpp
#ifndef ADIOSHELPERS_H
#define ADIOSHELPERS_H

#include <vector>
#include <iostream>

class AdiosHelpers {
public:
    static void define_adios_scalar(double scalar) {
        std::cout << "Defining ADIOS scalar with value: " << scalar << std::endl;
    }

    static void define_adios_global_real_1d_array(const std::vector<double>& array) {
        std::cout << "Defining ADIOS global real 1D array of size " << array.size() << std::endl;
        for (double val : array) {
            std::cout << val << " ";
        }
        std::cout << std::endl;
    }

    static void check_adios_err() {
        std::cout << "Checking for ADIOS errors..." << std::endl;
    }
};

#endif // ADIOSHELPERS_H